metricflow/dataset/convert_semantic_model.py

```diff
@@ -515,35 +515,55 @@ def create_sql_source_data_set(self, semantic_model: SemanticModel) -> SemanticM
         """Build data set for time spine."""
         from_source_alias = SequentialIdGenerator.create_next_id(StaticIdPrefix.TIME_SPINE_SOURCE).str_value
-        defined_time_granularity = time_spine_source.base_granularity
+        base_granularity = time_spine_source.base_granularity
         time_column_name = time_spine_source.base_column
 
         time_dimension_instances: List[TimeDimensionInstance] = []
         select_columns: List[SqlSelectColumn] = []
 
-        time_dimension_instance = self._create_time_dimension_instance(
-            element_name=time_column_name, entity_links=(), time_granularity=defined_time_granularity
+        # Build base time dimension instances & columns
+        base_time_dimension_instance = self._create_time_dimension_instance(
+            element_name=time_column_name,
+            entity_links=(),
+            time_granularity=ExpandedTimeGranularity.from_time_granularity(base_granularity),
         )
-        time_dimension_instances.append(time_dimension_instance)
-
-        dimension_select_expr = SemanticModelToDataSetConverter._make_element_sql_expr(
+        time_dimension_instances.append(base_time_dimension_instance)
+        base_dimension_select_expr = SemanticModelToDataSetConverter._make_element_sql_expr(
             table_alias=from_source_alias, element_name=time_column_name
         )
-        select_column = self._build_column_for_time_granularity(
-            time_granularity=defined_time_granularity,
-            expr=dimension_select_expr,
-            column_alias=time_dimension_instance.associated_column.column_name,
+        base_select_column = self._build_column_for_standard_time_granularity(
+            time_granularity=base_granularity,
+            expr=base_dimension_select_expr,
+            column_alias=base_time_dimension_instance.associated_column.column_name,
         )
-        select_columns.append(select_column)
-
-        new_instances, new_columns = self._build_time_dimension_instances_and_columns(
-            defined_time_granularity=defined_time_granularity,
+        select_columns.append(base_select_column)
+        new_base_instances, new_base_columns = self._build_time_dimension_instances_and_columns(
+            defined_time_granularity=base_granularity,
             element_name=time_column_name,
             entity_links=(),
-            dimension_select_expr=dimension_select_expr,
+            dimension_select_expr=base_dimension_select_expr,
         )
-        time_dimension_instances.extend(new_instances)
-        select_columns.extend(new_columns)
+        time_dimension_instances.extend(new_base_instances)
+        select_columns.extend(new_base_columns)
+
+        # Build custom granularity time dimension instances & columns
+        for custom_granularity in time_spine_source.custom_granularities:
+            custom_time_dimension_instance = self._create_time_dimension_instance(
+                element_name=time_column_name,
```

Comment:
Can we add a comment explaining why we use the time_column_name here? It feels odd as a reader, because it seems like we should be using the custom name, but from looking at the snapshots I think we need this so that the granularity is appended as it is elsewhere - i.e., we want `metric_time__martian_day` not `martian_day`, and that implies also `ds__martian_day` instead of `martian_day`.